### PR TITLE
feat: Add /unassign command to allow contributors to unassign themselves

### LINKED ISSUE #1239: [Intermediate]: Add `/unassign` command to allow contributors to unassign themselves
### Problem

The C++ SDK currently supports the /assign command to automatically assign a contributor and update labels (e.g., switching from status: ready for dev to status: in progress).

However, if a contributor later realizes they don't have the bandwidth to complete the issue, there is currently no self-service way to unassign themselves.

This can leave issues in a stale assigned state, requiring a maintainer to manually:

- remove the assignee
- revert the labels

This partially defeats the purpose of the automated workflow.

### Solution

Introduce a /unassign command that allows contributors to drop an issue they previously claimed.

When triggered, the workflow could:

1. Remove the commenter as an assignee
2. Remove the status: in progress label
3. Re-add the status: ready for dev label so others can claim it
4. Post a short acknowledgment comment

Contribution

If this sounds like a useful addition to the workflow, I’d be happy to open a PR implementing the /unassign command within the existing .github/scripts/commands setup.

### Alternatives

1) Manual unassignment by maintainers

Maintainers can currently remove assignees and update labels manually when a contributor is unable to continue working on an issue. However, this introduces additional manual work and can leave issues in an assigned state longer than necessary.

2) Automatic unassignment after inactivity

Another approach could be to automatically unassign contributors after a period of inactivity (for example, via a scheduled bot job). While this might help with stale assignments, it does not address the case where a contributor immediately realizes they cannot continue and wants to release the issue right away.

### PR DESCRIPTION
**Pull request by darshit2308**
**Description**:
Add support for the `/unassign` command to allow contributors to drop issues they no longer have bandwidth for, reducing manual maintainer overhead.

* Add `removeAssignees` helper to `helpers/api.js`
* Implement `/unassign` core logic (`commands/unassign.js`) validating issue state, authorization, and label reversion
* Create dedicated comment builders (`commands/unassign-comments.js`)
* Update `/assign` welcome comment to inform users about the new `/unassign` feature
* Hook `/unassign` route into the comment event dispatcher (`bot-on-comment.js`)

**Related issue(s)**:
Fixes #1239

**Notes for reviewer**:
* Tested locally using a new `test-unassign-bot.js` suite that mirrors the existing custom mock framework.
* The new tests provide full coverage for happy paths, authorization gates (including case-insensitive username matching), edge cases, and API failures.
* API label failures are logged server-side rather than silently failing to ensure system observability.

**Checklist**
- [x] Documented (Code comments)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/scripts/commands/unassign.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * 3. Is the commenter not the current assignee? -> unauthorized comment.
+ *
+ * On success: removes the user as an assignee, reverts the "in progress"
+ * label to "ready for dev", and posts an acknowledgment comment.
+ *
+ * @param {{ github: object, owner: string, repo: string, number: number,
+ * issue: object, comment: { user: { login: string } } }} botContext
+ * @returns {Promise<void>}
+ */
+async function handleUnassign(botContext) {
```
Comment (rwalworth) [inline .github/scripts/commands/unassign.js]: **suggestion (non-blocking):** The `/assign` handler calls `acknowledgeComment` as its first action to give the user immediate visual feedback (thumbs-up reaction on their comment). Consider adding the same here for consistency - it's a nice UX touch that signals the bot received the command before the heavier API calls run.
